main.go

```diff
@@ -307,7 +307,7 @@ func validateParams() providerConfig {
 	var ecrEnabled bool
 
 	awsAccountID = os.Getenv("awsaccount")
-	if len(awsAccountID) == 0 {
+	if len(awsAccountID) == 0 || awsAccountID == "changeme" {
```

Comment:
Yes, I think that merging the secrets is the best approach as all secrets are required for the pod to load.  I am debating what is the best way to configure the "mode" though.  A config map perhaps?